src/Symfony/Bridge/Doctrine/Middleware/AutoClose/AutoCloseDriver.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bridge\Doctrine\Middleware\AutoClose;
+
+use Doctrine\DBAL\Driver;
+use Doctrine\DBAL\Driver\Connection as DriverConnection;
+use Doctrine\DBAL\Driver\Middleware\AbstractDriverMiddleware;
+
+final class AutoCloseDriver extends AbstractDriverMiddleware
+{
+    public function __construct(
+        Driver $driver,
+        private \ArrayObject $connectionExpiries,
+        private readonly array $timeToLiveByConnection,
+        private readonly string $connectionName
+    ) {
+        parent::__construct($driver);
+    }
+
+    public function connect(array $params): DriverConnection
+    {
+        $connection = parent::connect($params);
+        $this->connectionExpiries[$this->connectionName] = time() + $this->timeToLiveByConnection[$this->connectionName];
```

Comment:
```suggestion
        $timestamp = time();
        $connection = parent::connect($params);
        $this->connectionExpires[$this->connectionName] = $timestamp + $this->timeToLiveByConnection[$this->connectionName];
```

Storing the current time before the connection will prevent weird race conditions (we will be sure that the expiration will be the TTL max. Currently, depending on the time it takes to do post-connect things, it can exceed the TTL.